shared/devices/index.render.desktop.js

```diff
@@ -1,160 +1,193 @@
-// TODO flow type this file
+/* @flow */
 import React, {Component} from 'react'
+import {Box, Text, Icon} from '../common-adapters'
+import {globalStyles, globalColors} from '../styles/style-guide'
+import type {Props as IconProps} from '../common-adapters/icon'
 
-import HardwarePhoneIphone from 'material-ui/lib/svg-icons/hardware/phone-iphone'
-import HardwareComputer from 'material-ui/lib/svg-icons/hardware/computer'
-import CommunicationVpnKey from 'material-ui/lib/svg-icons/communication/vpn-key'
-import ActionNoteAdd from 'material-ui/lib/svg-icons/action/note-add'
+import type {Props} from './index'
+import type {Device as DeviceType} from '../constants/types/flow-types'
 
-import moment from 'moment'
-import commonStyles from '../styles/common'
+class RevokedHeader extends Component {
+  state: {
+    expanded: boolean
+  };
 
-export default class DevicesRender extends Component {
-  renderPhone (device) {
-    return (
-      <div key={device.deviceID} style={{...styles.deviceOuter, ...styles.deviceShow}}>
-        <div style={styles.device}>
-          <HardwarePhoneIphone style={styles.deviceIcon}/>
-          <h3 style={styles.line2}>{device.name}</h3>
-          <div>Last used {moment(device.cTime).format('MM/DD/YY')}</div>
-          <div style={styles.line2}>TODO: Get Added info</div>
-          <p style={{...commonStyles.clickable, textDecoration: 'underline'}} onClick={() => this.props.showRemoveDevicePage(device)}>Remove</p>
-        </div>
-      </div>
-    )
+  constructor (props) {
+    super(props)
+    this.state = {
+      expanded: false
+    }
+    this.toggleHeader = this.toggleHeader.bind(this)
```

Comment:
we haven't been doing this up until now. there's some performance pluses but we haven't been caching these calls. intentional? also you can just use an arrow instead of bind